int8 init
int8 proc